### PR TITLE
Add risky orders push notifications

### PR DESCRIPTION
**Pull request by joiggama**
- [x] ~~Detect risky payments based on spree order risky detection logic~~
- [x] Add tests
- [x] Detect risky orders hooking into the state transition to complete
- [x] Update README to add push notifications configuration instructions.

### CODE REVIEW

File: app/models/spree/payment_decorator.rb [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+  def is_risky?
+    is_cvv_risky? || is_avs_risky? || state == "failed"
+  end
+
+  private
+
+  def send_risk_notification
+    SpreeSam::Notifications.build(:parse,
+      channels: (ENV["PARSE_CHANNELS"] || []).split(","),
```
Comment (joiggama) [inline app/models/spree/payment_decorator.rb]: Not so sure about this parse `channels` approach.

File: app/models/spree/payment_decorator.rb [hunk 2]
```diff
@@ -0,0 +1,29 @@
+Spree::Payment.class_eval do
+
+  after_save :send_risk_notification, if: :is_risky?
+
+  def is_risky?
+    is_cvv_risky? || is_avs_risky? || state == "failed"
```
Comment (softr8) [inline app/models/spree/payment_decorator.rb]: what if the credit card gets rejected by payment gateway? it'd change to 'failed' sending the notification, doesn't it?

File: app/models/spree/payment_decorator.rb (hunk 1 above)
Comment (softr8) [inline app/models/spree/payment_decorator.rb]: You can leverage Spree::Config object

File: app/models/spree/payment_decorator.rb [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+  end
+
+  private
+
+  def send_risk_notification
+    SpreeSam::Notifications.build(:parse,
+      channels: (ENV["PARSE_CHANNELS"] || []).split(","),
+      data: {
```
Comment (softr8) [inline app/models/spree/payment_decorator.rb]: I feel this object should go in Provider::Parse implementation

File: app/models/spree/payment_decorator.rb (hunk 2 above)
Comment (joiggama) [inline app/models/spree/payment_decorator.rb]: Yes, according to [`Spree::Order#is_risky?`](https://github.com/spree/spree/blob/3-0-stable/core/app/models/spree/order.rb#L499-L501), orders riskiness is defined by [`Spree::Payment.risky`](https://github.com/spree/spree/blob/3-0-stable/core/app/models/spree/payment.rb#L59) scope which considers the `failed` state as a condition to make the payment, and order risky.

**Comment by joiggama** [comment]
@softr8 Updated to hook into order transition to complete.

File: app/models/spree/order_decorator.rb [hunk 4]
```diff
@@ -21,4 +21,24 @@
...
-end
\ No newline at end of file
+  state_machine do
+    after_transition to: :complete, do: :send_risk_notification, if: :is_risky?
+  end
+
+  private
+
+  def send_risk_notification
+    SpreeSam::Notifications.build(:parse,
```
Comment (softr8) [inline app/models/spree/order_decorator.rb]: is :parse the backend? if so, what if it's moved to the configurator so you can switch easily between pusher services

**Comment by softr8** [comment]
Besides my comment, LGTM!

File: app/models/spree/order_decorator.rb (hunk 4 above)
Comment (joiggama) [inline app/models/spree/order_decorator.rb]: Yes, that makes a lot of sense.

**Comment by joiggama** [comment]
> Besides my comment, LGTM!

Done, made the notification provider a preference that defaults to parse provider.

**Pull request merged**